src/algebra/support.lean

```diff
@@ -309,35 +293,26 @@ by rw [mul_support_comp_eq_preimage f g, image_inter_preimage]
 end set
 
 namespace pi
-variables {A : Type*} {B : Type*} [decidable_eq A] [has_zero B] {a : A} {b : B}
+variables {A : Type*} {B : Type*} [decidable_eq A] [has_one B] {a : A} {b : B}
+
+open function
 
-lemma support_single_zero : function.support (pi.single a (0 : B)) = ∅ := by simp
+@[to_additive] lemma mul_support_mul_single_subset : mul_support (mul_single a b) ⊆ {a} :=
+λ x hx, by_contra $ λ hx', hx $ mul_single_eq_of_ne hx' _
 
-@[simp] lemma support_single_of_ne (h : b ≠ 0) :
-  function.support (pi.single a b) = {a} :=
-begin
-  ext,
-  simp only [mem_singleton_iff, ne.def, function.mem_support],
-  split,
-  { contrapose!,
-    exact λ h', single_eq_of_ne h' b },
-  { rintro rfl,
-    rw single_eq_same,
-    exact h }
-end
+@[to_additive] lemma mul_support_mul_single_one : mul_support (mul_single a (1 : B)) = ∅ :=
+by simp
 
-lemma support_single [decidable_eq B] :
-  function.support (pi.single a b) = if b = 0 then ∅ else {a} := by { split_ifs with h; simp [h] }
+@[simp, to_additive] lemma mul_support_mul_single_of_ne (h : b ≠ 1) :
+  mul_support (mul_single a b) = {a} :=
+mul_support_mul_single_subset.antisymm $
+  λ x (hx : x = a), by rwa [mem_mul_support, hx, mul_single_eq_same]
 
-lemma support_single_subset : function.support (pi.single a b) ⊆ {a} :=
-begin
-  classical,
-  rw support_single,
-  split_ifs; simp
-end
+@[to_additive] lemma mul_support_mul_single [decidable_eq B] :
+  mul_support (mul_single a b) = if b = 1 then ∅ else {a} := by { split_ifs with h; simp [h] }
 
-lemma support_single_disjoint {b' : B} (hb : b ≠ 0) (hb' : b' ≠ 0) {i j : A} :
-  disjoint (function.support (single i b)) (function.support (single j b')) ↔ i ≠ j :=
-by rw [support_single_of_ne hb, support_single_of_ne hb', disjoint_singleton]
+lemma mul_support_mul_single_disjoint {b' : B} (hb : b ≠ 1) (hb' : b' ≠ 1) {i j : A} :
```

Comment:
Is there a reason why there is no `to_additive` here?

Reply:
```suggestion
@[to_additive]
lemma mul_support_mul_single_disjoint {b' : B} (hb : b ≠ 1) (hb' : b' ≠ 1) {i j : A} :
```